combat-trainer.lic

```diff
@@ -904,6 +905,34 @@ class SafetyProcess
 
   private
 
+  def check_item_recovery(game_state)
+    if Flags['ct-germshieldlost']
+      recover_item(game_state, Flags['ct-germshieldlost'].last, true)
+      Flags.reset('ct-germshieldlost')
+    end
+  end
+
+  def recover_item(game_state, item, wear = false)
+    return unless item
+    game_state.sheath_whirlwind_offhand
+    case bput("get #{item}", 'You pick up', 'What were you', 'too injured')
+    when 'What were you', 'too injured'
+      warn_failed_item_recovery
+    when 'you pick up'
+      if wear
+        bput("wear my #{item}", 'You sling', 'You attach', 'You .* unload', 'You strap', 'You slide', 'You work your way into', 'You spin', 'You put', 'You carefully loop', 'slide effortlessly onto your', 'You slip', "^You .* #{item.short_regex}")
+        waitrt?
+      else
+        warn_failed_item_recovery unless DRCI.put_away_item?(item)
+      end
+    end
+    game_state.wield_whirlwind_offhand
+  end
+
+  def warn_failed_item_recovery
+    5.times{ DRC.message "YOU HAVE DROPPED YOUR SHIELD AND WE CANNOT RECOVER!"; beep }
+  end
```

Comment:
These updates look great. One tweak I suggest is pass the item to this item recovery method so it can be used in the message instead of the hard coded word “SHIELD”. It’ll make it more clear in the log what went missing and the method more reusable. Thanks

Reply:
Oh right, I expanded it out from just shield. Thank you.